webpack/JobWizard/__tests__/integration.test.js

```diff
@@ -0,0 +1,138 @@
+import { mount } from '@theforeman/test';
+import * as api from 'foremanReact/redux/API';
+import { JobWizard } from '../JobWizard';
+import * as selectors from '../JobWizardSelectors';
+
+jest.spyOn(api, 'get');
+jest.spyOn(selectors, 'selectJobTemplate');
+jest.spyOn(selectors, 'selectJobTemplates');
+jest.spyOn(selectors, 'selectJobCategories');
+jest.spyOn(selectors, 'selectJobCategoriesStatus');
+
+const jobCategories = ['Ansible Commands', 'Puppet', 'Services'];
+
+api.get.mockImplementation(({ handleSuccess, ...action }) => {
+  if (action.key === 'JOB_CATEGORIES') {
+    handleSuccess && handleSuccess({ data: { job_categories: jobCategories } });
+  } else if (action.key === 'JOB_TEMPLATE') {
+    handleSuccess &&
+      handleSuccess({
+        data: {
+          job_template: {
+            execution_timeout_interval: 2,
+          },
+          effective_user: {
+            value: 'default effective user',
+          },
+        },
+      });
+  }
+  return { type: 'get', ...action };
+});
+
+selectors.selectJobTemplate.mockImplementation(() => null);
+selectors.selectJobCategories.mockImplementation(() => jobCategories);
+selectors.selectJobCategoriesStatus.mockImplementation(() => null);
+selectors.selectJobTemplates.mockImplementation(() => jobTemplates);
+
+const jobTemplates = [
+  {
+    id: 178,
+    name: 'Run Command - Ansible Default',
+    job_category: 'Ansible Commands',
+    provider_type: 'Ansible',
+    snippet: false,
+    description_format: 'Run %{command}',
+  },
+];
+const jobTemplate = {
+  job_template: {
+    id: 178,
+    name: 'Run Command - Ansible Default',
+    template:
+      "---\n- hosts: all\n  tasks:\n    - shell:\n        cmd: |\n<%=       indent(10) { input('command') } %>\n      register: out\n    - debug: var=out",
+    snippet: false,
+    default: true,
+    job_category: 'Ansible Commands',
+    provider_type: 'Ansible',
+    description_format: 'Run %{command}',
+    execution_timeout_interval: null,
+    description: null,
+  },
+  effective_user: {
+    id: null,
+    job_template_id: 178,
+    value: null,
+    overridable: true,
+    current_user: false,
+  },
+};
+
+const mockStore = configureMockStore([]);
+const store = mockStore({ effective_user: { overridable: true } });
+describe('Job wizard fill', () => {
+  it('should select template', () => {
+    const wrapper = mount(
+      <Provider store={store}>
+        <JobWizard advancedValue={{}} setAdvancedValue={jest.fn()} />
+      </Provider>
+    );
+    expect(wrapper.find('.pf-c-wizard__nav-link.pf-m-disabled')).toHaveLength(
+      4
+    );
+    selectors.selectJobCategoriesStatus.mockImplementation(() => 'RESOLVED');
+    expect(store.getActions()).toMatchSnapshot('initial');
+
+    selectors.selectJobTemplate.mockImplementation(() => jobTemplate);
+    wrapper.find('.pf-c-button.pf-c-select__toggle-button').simulate('click');
+    wrapper.find('.pf-c-select__menu-item').simulate('click');
```

Comment:
from what I remember from past implementation, sometimes something breaks when enzyme or jest upgrades and these clicks won't always work.. I would suggest triggering the actual action, e.g: `wrapper.instance().handleClick()`

Reply:
I tried using `wrapper.instance()` but it returns null, because:  `With React 16 and above, instance() returns null for stateless functional components.`
https://enzymejs.github.io/enzyme/docs/api/ReactWrapper/instance.html
What can cause `wrapper.find('.pf-c-select__menu-item').simulate('click');` to break?

Reply:
Alright, but you can use `wrapper.props()`? if I recall and then trigger the method you wanted
yes, also if the DOM changes a bit it can break your test while the button is still there...
anyway it's a better practice to trigger the actual method you want